### DISCUSSION TITLE
peaksPerCell parameter for addReproduciblePeakSet()

### DISCUSSION THREAD

**Discussion by mzhibo** [GreenleafLab/ArchR#1082] [Questions / Documentation] [answered]
Hi,
I have a question on the `peaksPerCell` parameters in `addReproduciblePeakSet() `. 
Why `peaksPerCell` is set to 500 while the `maxPeaks` is set to 150k. If I understand it correctly, this makes it only keep up to 500 peaks per group. macs2 cutoff for a p or q value would select significant peaks.  In addition,  the coverage file is generated with similar number of cells by subsampling. I wonder why setting an upper limit of peaksPerCells here. Would this force to drop some true peaks in groups that have a good number of cells (~1k)?
For clusters that all have above 1k cells, what's your recommended value for peaksPerCell? If I set peaksPerCell to NULL, will it just rely on the cutoff threshold?
Thank you
Zhibo

**Comment by rcorces** [comment, marked as answer]
Sorry - that interpretation is not correct. It is not `peaksPerCellGrouping`, it is `peaksPerCell`. If you have `peaksPerCell = 500` and your cell grouping has 100 cells, then the maximum number of peaks that grouping could have would be `500*100` or `50,000`
As the parameter says, this is to prevent groups with very few cells from contributing lots of peaks

**Reply by rcorces** [reply]
also `peaksPerCell` must be an integer, it cannot be `NULL`

**Reply by mzhibo** [reply]
Thank you for clarifying!
